keps/sig-node/693-topology-manager/README.md

```diff
@@ -594,49 +593,305 @@ feature gate and set the CPU Manager policy to 'static'.
+resources aligned by Topology Manager are running on the node, disabling
+feature gate won't cause any issue.
+
+If the feature gate is being disabled on a node where such pods are running,
+it is the responsibliity of the cluster admin to ensure that the node is
+appropriately drained.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+No impact on running pods in the cluster. Subsequent pods that meet the requirement
+as explained above would go through admission check and aligned based on the
+configured policy.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, covered by node e2e tests:
+
+https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/topology_manager_test.go
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollout can fail in case a bug is introduced in topology manager preventing already
+running pods from restarting or new pods to start.
+
+###### What specific metrics should inform a rollback?
+
+`topology_manager_admission_errors_total` can be used to determine the health of the
+feature and can be be used to determine if a rollback needs to be performed. It is
+worth noting that there could be valid cases where a pod is denied admission.
+
+Example of a valid error is `TopologyAffinityError` which is returned when topology
+manager is configured with `single-numa-node` but there are not enough resources
+available on a single numa node. 
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No. Changes in behavior only affects pods meeting the conditions scheduled after the upgrade.
+Running pods will be unaffected by any change. This offers some degree of safety in both upgrade->rollback and upgrade->downgrade->upgrade scenarios.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
 
-* *TBD*
+Monitor the following metrics:
+
+"topology_manager_admission_requests_total"
+"topology_manager_admission_errors_total"
+"topology_manager_admission_duration_seconds"
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The operator can look at `topology_manager_admission_requests_total`, `topology_manager_admission_errors_total` and
+`topology_manager_admission_duration_seconds` metrics to determine if topology manager is performing its admission check.
+In addition to that, kubelet configuration of the nodes can be inspected to check feature gates and the policies
+configured.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [X] Other (treat as last resort)
+  - Details: check the kubelet metric `topology_manager_admission_requests_total` or "topology_manager_admission_duration_seconds"
```

Comment:
I understand that you want me to populate this section from end-user perspective. Thanks for pointing that out.

By design, NUMA information is hidden from the end users and is only known to kubelet running on the node. In order to validate that the allocated resources are NUMA aligned, we need this information to be exposed. The only possible way is with the help of external tools that inspect the resource topology information and either expose it external to the node (e.g. [NFD topology updater](https://github.com/kubernetes-sigs/node-feature-discovery/blob/master/docs/get-started/introduction.md#nfd-topology-updater)) or use it to perform validation themselves ([numaalign](https://github.com/ffromani/numalign)).

Here are a few possible options (with external help):

1. In case Topology manger is configured with `single-numa-node` policy and CPU Manager with `static policy`  Using NFD topology updater, we learn about the number of allocatable CPUs on a NUMA node and deploy a pod with CPUs greater than we have available on a single NUMA node. In that case, the pod would return a `TopologyAffinityError` and is visible to the end user.
2. Alternatively, we can use a tool like [numaalign](https://github.com/ffromani/numalign) and run that within a pod to determine if a set of resources is aligned on the same NUMA node.

Kubernetes on its own without any external help does not have any mechanism to validate the correct operation of this feature and that is by design unfortunately. I would appreciate your input on how to handle this for this KEP.

Reply:
I haven't gone back an swapped in all the context, so forgive me if this makes no sense. Without doing that, my guess is the UX for this feature is either:1) node labels and selectors to schedule to a node meeting the topology policy requirements; 2) an explicit way to state that requirement in your pod template. Conceptually, allowing users to specify a requirement (e.g., state "i need this alignment policy"), without allowing them to validate that it is being met is an incomplete feature. So if 2) is the way this is done, I would ask for us to eventually plumb validation of this all the way. If you are using 1) as the methodology, we are sort of delegating responsibility downward and it's not as critical to plumb it all the way through - in other words in that case we don't strictly need k8s to expose this directly.

All that said, none of it would block this PR.

For this particular question, what we want is guidance for a user that is debugging or otherwise troubleshooting. So, I am ok with "here's how I would check it" using external tools. So probably just take what you said in this discussion and adapt to answer the question, that will be great.

The rest of the discussion is for future consideration around the UX of this feature for the SIG.